scripts/ansible/roles/linux/intel/tasks/redhat/sgx-packages-install.yml

```diff
@@ -44,10 +44,38 @@
   register: install_sgx
   until: install_sgx is success
 
-- name: Install the Intel DCAP packages
+- name: Download Intel SGX DCAP Packages
+  get_url:
+    url: "{{ intel_dcap_packages_url }}"
+    dest: "/tmp/sgx_linux_dcap_packages.tgz"
+    mode: 0755
+    timeout: 120
+  retries: 3
+  when: flc_enabled|bool
+
+- name: Create Intel DCAP directory
+  file:
+    path: "{{ intel_dcap_packages_directory }}"
+    state: directory
+
+- name: Unarchive Intel DCAP local rpm repo tgz
+  unarchive:
```

Comment:
See the unarchive module's `create` option, which is useful for determining if the step needs to be ran or not given the existence of a file path. https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html
For example:
```
- name: Unarchive Intel DCAP local rpm repo tgz
  unarchive:
    src: "/tmp/sgx_linux_dcap_packages.tgz"
    dest: "{{ intel_dcap_packages_directory }}"
    remote_src: yes
    creates: "{{ intel_dcap_packages_directory }}/sgx_rpm_local_repo"
```

Reply:
Instead of creating only if the file path doesn't exist, I've added another step to delete the directory if it does exist.